runhouse/resources/hardware/on_demand_cluster.py

```diff
@@ -528,16 +541,27 @@ def up(self, verbose: bool = True, force: bool = False):
         if self.launcher_type == LauncherType.DEN:
             logger.info("Launching cluster with Den.")
             launcher = DenLauncher(cluster=self, force=force)
+            data = launcher.up(verbose=verbose)
+            self._update_from_den_response(data)
+
         else:
             logger.info("Launching cluster locally.")
             launcher = LocalLauncher(cluster=self)
+            launcher.up(verbose=verbose)
+            self._update_from_sky_status()
```

Comment:
we should probably call this inside the launcher too after the cluster has been upped and before returning to den